### ISSUE TITLE
"Parameter/variable is not used" false positive with assignop and bitwise AND

### ISSUE THREAD

**Issue by Daimona** [kalessil/phpinspectionsea#1816]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.7.1 |
| Language level | PHP 7.2                        |

Current behaviour
--
For the following code:
```php
<?php

function test( int $x ): int {
	$val = 0;
	$val |= 1; // Line 5
	return $x & $val;
}
```

at line 5, I get the following warning:
> Parameter/variable is overridden, but is never used or appears outside of the scope.

Expected behaviour
--

No warning, the variable is clearly used on the next line.

With other operators (e.g. `|`, `^`, `&&`, etc.) it works as expected. Instead, it remains broken if I change the assignop (tested `&=`, `+=` and `.=`).

Environment details
---

PhpStorm 2021.3.2
Build #PS-213.6777.58, built on January 31, 2022

**Comment by Daimona** [comment]
> But isn't this exactly the same as [...]

Sure, but mine was just a simplified example. You can replace `$val = 0` with `$val = rand()` (or anything else) and you'll still get a warning.
